### PR TITLE
refactor(controller)!: remove possibility of separate rollouts manager/client

### PR DESCRIPTION
**Pull request by krancour**
Since we've become committed to AnalysisRuns used for verification purposes always living within the Kargo control plane, the door is now closed on the possibility of Kargo controllers using a separate client for communication with a different cluster that hosts AnalysisRun resources.

This PR removes code that was already rendered dead by that decision.

Note: This _doesn't_ mean AnalysisRuns cannot _execute_ elsewhere now. That continues to be achievable by setting `controller.rollouts.controllerInstanceID` on a Kargo controller.

cc @jessesuen and @gdsoumya

### CODE REVIEW

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 1]
```diff
@@ -93,26 +93,4 @@ rules:
...
-metadata:
-  name: kargo-controller-rollouts
-  labels:
-    {{- include "kargo.labels" . | nindent 4 }}
-    {{- include "kargo.controller.labels" . | nindent 4 }}
-rules:
-- apiGroups:
-  - argoproj.io
-  resources:
-  - analysisruns
```
Comment (hiddeco) [inline charts/kargo/templates/controller/cluster-roles.yaml:109]: I think these permissions are still required by the verification process?

Comment (krancour) [reply charts/kargo/templates/controller/cluster-roles.yaml:109]: Odd... this all _worked_. Let me look into that.

Comment (krancour) [reply charts/kargo/templates/controller/cluster-roles.yaml:109]: Oh... I removed those last minute _after_ I saw it working. Fixing.

**Review by hiddeco — APPROVED** [review]
🚀

**Pull request merged**